hivemind/server/__init__.py

```diff
@@ -113,8 +117,36 @@ def create(listen_on='0.0.0.0:*', num_experts=None, expert_uids=None, expert_cls
         # initialize experts
         if expert_uids is None:
             num_experts = num_experts if num_experts is not None else 1
-            expert_uids = [f'{expert_prefix}{hivemind.DHT.UID_DELIMITER}{i + expert_offset}'
-                           for i in range(num_experts)]
+            expert_pattern = expert_pattern if expert_pattern is not None else f"expert.[0:{num_experts}]"
+            print(expert_pattern)
+
+            # parse expert pattern
+            expert_pattern_blocks = expert_pattern.split(hivemind.DHT.UID_DELIMITER)
+
+            def generate_uid():
+                uid = ''
+                for block in expert_pattern_blocks:
+                    try:
+                        if '[' not in block and ']' not in block:
+                            uid += block
```

Comment:
String concatenation is not free, better to keep them in a list and use .join() afterwards

Reply:
thx, fixed